torch/functional.py

```diff
@@ -1988,6 +1988,12 @@ def _unravel_index(indices: Tensor, shape: Union[int, Sequence[int]]) -> Tensor:
             )
         )
     )
+
+    torch._check_value(
+        all(index < coefs[0] * shape[0] for index in indices),
+        lambda: f"'indices' must be in range {[0, coefs[0] * shape[0] - 1]}, but got {indices} instead.",
+    )
```

Comment:
This is exactly why we avoid checking indices. The problem is that if tensors live on the GPU, the `all` will cause a CPU-GPU synchronization, which can be expensive. Is the check missing in the kernel? If so, it is best to be placed there.